### DISCUSSION TITLE
Code folding for extended languages

### DISCUSSION THREAD

**Discussion by jtlandis** [zed-industries/zed#16978] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Hello Zed devs! I'm loving my experience with Zed so far!

I was wondering how we would implement code folding for zed for extended languages?

Is this specified by the LSP, or is this something that could be handled with tree-sitter and some @collapsable  capture group?

**Comment by notpeter** [comment]
I believe this is done entirely using tree-sitter.  You can see this in action by looking at an extension which does not bundle an LSP binary (e.g. [zed-nix-extension](https://github.com/zed-extensions/nix)) and see folding still works as expected.

**Comment by jtlandis** [comment]
I am unsure if code folding is truly handled by tree-sitter. Opening a blank file in which "Select Language" says "Unknown", you can still see code folding suggestions in the gutter. To get code folding, it seems to require that the parent as some type of indenting?

```
{
  test
}

* test 1
  * sub test 1
* test 2
```

Being controlled by indenting makes sense for a majority of languages, but I was hoping to allow for control of collapsing of markdown headers. I could put capture for indenting, but folding would only work if the sub section was actually indented.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
